paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.cc

```diff
@@ -52,36 +52,25 @@ bool HasBias(ir::Node* conv_op) {
          conv_op->Op()->Input("Bias").size() > 0;
 }
 
-bool ShouldSkipConv(ir::Node* conv_op, Scope* scope, ir::Node* conv_filter) {
-  if (!platform::HasOpINT8DataType(conv_op->Op())) {
-    VLOG(4) << "Skipping non-int8 convolution (id: " << conv_op->id() << ").";
-    return true;
-  }
-
-  auto filter_var = scope->GetVar(conv_filter->Name());
-  if (filter_var->Get<LoDTensor>().dtype() != phi::DataType::FLOAT32) {
-    VLOG(4) << "Skipping convolution (id: " << conv_op->id()
-            << ") because it's a bug that it is detected again.";
-    return true;
-  }
-
-  VLOG(4) << "Not skipping convolution (id: " << conv_op->id() << ")";
-  return false;
-}
-
 template <typename T>
 void QuantizeConvInput(Scope* scope,
                        ir::Graph* g,
                        ir::Node* conv_op,
                        const std::string& input_name,
                        const std::string& scales_attr_name) {
-  const auto scales =
-      conv_op->Op()->GetAttrIfExists<std::vector<float>>(scales_attr_name);
-
-  auto* tensor = scope->GetVar(input_name)->GetMutable<LoDTensor>();
-  QuantizeParams<T>(tensor, scales);
-
-  conv_op->Op()->SetAttr(scales_attr_name, std::vector<float>(1, 1));
+  auto var = scope->GetVar(input_name);
+  if (var->Get<LoDTensor>().dtype() != phi::DataType::FLOAT32) {
+    VLOG(0) << "Skipping convolution filter: " << input_name
+            << " because it is detected again.";
```

Comment:
QuantizeConvInput is called for both filters and biases so this message might be printed twice now and will say filter even when it may be bias. The reason for this if check was to quit early if the same conv pattern was detected twice. Now the handler code might be executed twice and it will just skip only quantization of weights and biases but in the end the counter will show it as quantized so the LOG will say that more conv ops were quantized than actually exist in the model.

Reply:
Done

Reply:
Maybe I was wrong about the second part. Maybe it's not detected twice. The ConvResidual pattern used might be working correctly as opposed to Conv pattern in int8_scale_calculation_mkldnn_pass. If so, it will display correct number of detected ops and just avoid quantizing them again. So it seems good to me.